frontend/apps/app/components/SessionDetailPage/types.ts

```diff
@@ -46,6 +46,10 @@ export type TimelineItem = Pick<
       executed_at: string
     }>
   } | null
+  // TODO: Backend needs to add artifact_action field to timeline_items table
+  // This field should be set to 'created' when PM agent creates requirements artifact
+  // and 'updated' when QA agent adds use cases to the artifact
+  artifact_action?: 'created' | 'updated' | null
```

Comment:
💭 I'm thinking it would be better to make the implementation compliant with LangChain's tool_calls, but it looks painful without removing timeline_items and pulling it closer to messages. I'm torn. I think it's best to keep it that way for now.